lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -93,13 +93,38 @@ def call(env)
             request_span.resource = nil
             request_trace = Tracing.active_trace
             env[Ext::RACK_ENV_REQUEST_SPAN] = request_span
+            Thread.current[:datadog_http_routing] = []
 
             # Copy the original env, before the rest of the stack executes.
             # Values may change; we want values before that happens.
             original_env = env.dup
 
             # call the rest of the stack
             status, headers, response = @app.call(env)
+
+            # TODO: might be wrong if say rails routed to sinatra but sinatra
+            #       fails to find a route, so I dumbly checked for 404?
+            if status != 404 && (routed = Thread.current[:datadog_http_routing].last)
```

Comment:
We should probably log `Thread.current[:datadog_http_routing]` because from what I've seen it can be very surprising at times and without this full thing nor the intermediate routing spans and tags it's absolutely impossible to debug anything.